src/taskgraph/transforms/run/__init__.py

```diff
@@ -11,162 +11,134 @@
-        Optional("run-on-git-branches"): task_description_schema["run-on-git-branches"],
-        Optional("shipping-phase"): task_description_schema["shipping-phase"],
-        Optional("always-target"): task_description_schema["always-target"],
-        Exclusive("optimization", "optimization"): task_description_schema[
-            "optimization"
-        ],
-        Optional("needs-sccache"): task_description_schema["needs-sccache"],
-        Exclusive(
-            "when",
-            "optimization",
-            description=dedent(
-                """
-                The "when" section contains descriptions of the circumstances under
-                which this task should be included in the task graph. This will be
-                converted into an optimization, so it cannot be specified in a run
-                description that also gives 'optimization'.
-                """
-            ),
-        ): {
-            Optional(
-                "files-changed",
-                description=dedent(
-                    """
-                    This task only needs to be run if a file matching one of the given
-                    patterns has changed in the push. The patterns use the mozpack
-                    match function (python/mozbuild/mozpack/path.py).
-                    """
-                ),
-            ): [str],
-        },
-        Optional(
-            "fetches",
-            description=dedent(
-                """
-                A list of artifacts to install from 'fetch' tasks.
-                """
-            ),
-        ): {
-            str: [
-                str,
-                fetches_schema,
-            ],
-        },
-        Required(
-            "run",
-            description=dedent(
-                """
-                A description of how to run this task.
-                """
-            ),
-        ): {
-            Required(
-                "using",
-                description=dedent(
-                    """
-                    The key to a run implementation in a peer module to this one.
-                    """
-                ),
-            ): str,
-            Optional(
-                "workdir",
-                description=dedent(
-                    """
-                    Base work directory used to set up the task.
-                    """
-                ),
-            ): str,
-            # Any remaining content is verified against that run implementation's
-            # own schema.
-            Extra: object,
-        },
-        Required("worker-type"): task_description_schema["worker-type"],
-        Optional(
-            "worker",
-            description=dedent(
-                """
-                This object will be passed through to the task description, with additions
-                provided by the task's run-using function.
-                """
-            ),
-        ): dict,
-    }
-)
+
+# Fetches schema using msgspec
+class FetchesStruct(Struct):
```

Comment:
Would you mind splitting all the actual schema conversion work into a separate commit from supporting msgspec in validate_schema? This way it will be easy to prove that the generation works before and after the change.